File: Javascript/backend_nest/src/dto/mapper/UserDTOMapper.ts
import { User } from 'src/domain/entities/User';
import { Mapper } from 'src/domain/kernel/Mapper';
import { UserDTO } from '../schema/UserDTO';

export class UserDTOMapper extends Mapper<User, UserDTO> {
  toDomain(external: UserDTO): User {
    return User.create({
      id: external?.id || null,
      email: external.email,
      role: external.role,
      firstName: external.firstName,
      lastName: external.lastName,
    });
  }

  toExternal(domain: User): UserDTO {
    return {
      id: domain.id,
      email: domain.email,
      role: domain.role,
      firstName: domain.firstName,
      lastName: domain.lastName,
    };
  }
}

File: Python/backend_fastapi/shared/dto/mappers/user_dto_mapper.py
from domain.kernel.Mapper import BaseMapper
from domain.entities.User import User
from shared.dto.schemas.user_dto import UserDTO

class UserDTOMapper(BaseMapper[User, UserDTO]):
    def to_domain(self, dto: UserDTO, user_id: int = 0) -> User:
        return User.create(
            id=user_id,
            email=dto["email"],
            role=dto["role"],
            first_name=dto["first_name"],
            last_name=dto["last_name"]
        )

    def to_external(self, user: User) -> UserDTO:
       return UserDTO(
            email=user.email,
            role=user.role,
            first_name=user.first_name,
            last_name=user.last_name
        )